src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -271,12 +368,30 @@ class ExternalDatabaseManagementService(
         updateHBARecords(dbName)
     }
 
+    fun getOrganizationOwners(orgId: UUID): Set<SecurablePrincipal> {
+        val principals = securePrincipalsManager.getAuthorizedPrincipalsOnSecurableObject(AclKey(orgId), EnumSet.of(Permission.OWNER))
+        return securePrincipalsManager.getSecurablePrincipals(principals).toSet()
+    }
+
     /**
      * Sets privileges for a user on an organization's table or column
      */
     fun executePrivilegesUpdate(action: Action, acls: List<Acl>) {
-        val aclsByOrg = acls.groupBy { it.aclKey[0] }
-        aclsByOrg.forEach { (orgId, orgAcls) ->
+        //split acls between table and column objects
+        val aclsByType = acls.partition { it.aclKey.size == 1 }
+
+        val tableAclsByOrg = aclsByType.first.groupBy {
+            organizationExternalDatabaseTables.getValue(it.aclKey[0]).organizationId
+        }
+        val columnAclsByOrg = aclsByType.second.groupBy {
+            organizationExternalDatabaseColumns.getValue(it.aclKey[1]).organizationId
+        }
+        val orgIds = tableAclsByOrg.keys.union(columnAclsByOrg.keys)
+
+        orgIds.forEach { orgId ->
+            val orgAcls = mutableListOf<Acl>()
+            if (tableAclsByOrg.isNotEmpty()) tableAclsByOrg.getValue(orgId).toMutableList()
```

Comment:
Also, you can just do `val orgAcls = tableAclsByOrg[orgId]?.toMutableList()` and then `columnAclsByOrg[orgId]?.let { orgAcls.addAll(it) }`

Reply:
Sorry, the first expression should be 
`val orgAcls = tableAclsByOrg[orgId]?.toMutableList() ?: mutableListOf()`